pkg/controller/worker/machine_dependencies.go

```diff
@@ -14,12 +14,281 @@
+			deviceID, err := deviceIDFromProviderID(n.Spec.ProviderID)
+			fmt.Printf("deviceID %s err %#v", deviceID, err)
+			if deviceID == "" || err != nil {
+				continue
+			}
+			nodePrivateNetwork, err := getNodePrivateNetwork(deviceID, c, ns)
+			if err != nil {
+				return fmt.Errorf("error getting private network from Equinix Metal API for %s: %v", n.Spec.ProviderID, err)
+			}
+			if nodePrivateNetwork == "" {
+				continue
+			}
+			if n.Annotations == nil {
+				n.Annotations = map[string]string{}
+			}
+			// if it was not set already, set it and save it
+			if n.Annotations[equinixMetalPrivateNetworkAnnotations] != nodePrivateNetwork {
+				n.Annotations[equinixMetalPrivateNetworkAnnotations] = nodePrivateNetwork
+				// update the node in the k8s cluster ***
+				patch, _ := json.Marshal(map[string]interface{}{
+					"metadata": map[string]interface{}{
+						"annotations": n.Annotations,
+					},
+				})
+				if err := shoot.Patch(ctx, &n, client.RawPatch(types.StrategicMergePatchType, patch)); err != nil {
+					return fmt.Errorf("unable to patch node %s with private network cidr: %v", n.Name, err)
+				}
+			}
+		}
+		targetCidrs = append(targetCidrs, n.Annotations[equinixMetalPrivateNetworkAnnotations])
+	}
+
+	// sort them for consistency; the order really doesn't matter, as long as it is consistent
+	targetCidrs = unique(targetCidrs)
+	sort.Strings(targetCidrs)
+	cidrs := strings.Join(targetCidrs, ",")
+
+	deploy := &appsv1.Deployment{}
+	if err := c.Get(ctx, client.ObjectKey{
+		Namespace: ns,
+		Name:      apiServerDeploy,
+	}, deploy); err != nil {
+		return fmt.Errorf("failed to get kube-apiserver deployment: %v", err)
+	}
+
+	// find the vpn-seed container
+	ctrs := deploy.Spec.Template.Spec.Containers
+	var changed bool
+	for i, ctr := range ctrs {
+		// find the right container
+		if ctr.Name != vpnSeed {
+			continue
+		}
+		// find the right env var
+		var found bool
+		for _, env := range ctr.Env {
+			if env.Name != nodeNetworkEnvVar {
+				continue
+			}
+			// track that it existed
+			found = true
+			if env.Value != cidrs {
+				env.Value = cidrs
+			}
+		}
+		// if we did not find it, so add it
+		if !found {
+			ctr.Env = append(ctr.Env, corev1.EnvVar{Name: nodeNetworkEnvVar, Value: cidrs})
+		}
+		ctrs[i] = ctr
+		changed = true
+	}
+	if !changed {
+		return nil
+	}
+	return c.Update(ctx, deploy)
+}
+
+func (w *workerDelegate) UpdateConfigMap(ctx context.Context) error {
+	ns := w.worker.GetNamespace()
+	c := w.Client()
+
+	// get the private IPs from the shoot nodes
+	_, shoot, err := util.NewClientForShoot(ctx, c, ns, client.Options{})
+	if err != nil {
+		return err
+	}
+
+	shootNodes := &corev1.NodeList{}
+	if err := shoot.List(ctx, shootNodes); err != nil {
+		return fmt.Errorf("failed to get shoot nodes: %v", err)
+	}
+
+	// go through each node, for each one without the right annotation, get the private network
+	var targetCidrs []string
```

Comment:
You could use `sets.String` - it also allows you to delete the `unique` function and its `List()` method implicitly sorts

Reply:
Which?

Reply:
https://github.com/kubernetes/apimachinery/blob/57f2a0733447cfd41294477d833cce6580faaca3/pkg/util/sets/string.go

Reply:
Eh, I don't care too much. It works.

Reply:
Fair enough, just a suggestion. Resolving this.